src/Backend/DataSeriesFactory.js

```diff
@@ -2,7 +2,19 @@ export default class TraceFactory {
     static getXValuesByPoints(minValue, maxValue, numberOfPoints) {
         if(numberOfPoints === 1) return [(maxValue + minValue)/2.0]
         const interval = (maxValue - minValue) / (numberOfPoints - 1)
-        return Array.from(new Array(~~numberOfPoints), (_, key) => { return minValue + interval * key })
+        return Array.from(new Array(numberOfPoints), (_, key) => { return minValue + interval * key })
+    }
+
+    static getLogXValuesByPoints(minValue, maxValue, numberOfPoints){
+        const values = new Array(numberOfPoints);
```

Comment:
if number of points is 1, we should return geometric mean of minValue and maxValue, namely sqrt( minValue * maxValue)